src/event_handler.rs

```diff
@@ -19,6 +19,45 @@ use std::error::Error;
 use std::process::{Command, Stdio};
 use std::time::Instant;
 
+#[derive(Debug, Clone, Copy)]
+pub struct MappableEvent {
+    key: Key,
+    value: i32,
+    underlying: Option<InputEvent>,
+}
+
+impl MappableEvent {
+    pub fn new(key: Key, value: i32) -> Self {
+        Self { key, value, underlying: None }
+    }
+
+    pub fn from(event: InputEvent) -> Option<Self> {
+        match event.kind() {
+            InputEventKind::Key(key) => {
+                Some(Self { key, value: event.value(), underlying: Some(event) })
+            },
+            InputEventKind::RelAxis(RelativeAxisType::REL_WHEEL) => {
+                let key = match event.value() {
+                    1 => Key::KEY_SCROLLUP,
+                    -1 => Key::KEY_SCROLLDOWN,
+                    other => {
+                        debug!("Unknown scroll value: {}", other);
+                        return None
+                    }
+                };
```

Comment:
Bug: Depending on how fast you scroll, it is possible to create scroll events with values higher than 1 or lower than -1, which would cause scroll events to trigger the "other" branch of the match pattern.
I'd suggest replacing 1 and -1 with 1..=i32::MAX and i32::MIN..=-1 or something similar.

Reply:
That seems like a reasonable change, though I am surprised. In my testing I only ever saw 1/-1, other values only occurred in the HIGHRES event variants. I've only got a wired mouse though, are you using a trackpad?

Reply:
No, i'm using a wired mouse too (PERIDUO-406 wired mouse).
I do need to scroll somewhat fast to get values higher/lower than 1/-1, though.